Templates (#856)
casper-network/docs: .github/templates/tutorial-template.md

<!-- This is a template to give you some rough structure for writing a tutorial.  -->
<!-- Using a template means anyone can contribute and the experience for users reading the tutorials remains as consistent as possible. -->

> <!-- However, feel free to delete anything that isn't relevant to your tutorial.  -->

Reviewer: ```suggestion
<!-- However, feel free to delete anything irrelevant to your tutorial.  -->
```

> <!-- If you need any help, tag our developer advocates (info here) or TW team (info here). Thank you for contributing! -->

Reviewer 2: We'll need to determine contact links for dev advocates/tech writers when we begin using this template extensively.

As is, I think it's okay to leave it like this until those determinations take place.

> # Tutorial title

Reviewer: We capitalize all words in level 1 (#) and level 2 (##) headers. If you'd like to change this, we need to agree and update all the headers in the docs.

Author: Oh no that's my mistake - apologies

## Overview

> Give a high-level overview of the tutorial. This helps the reader understand the scope of what they'll be doing e2e and potentially the basic architecture of what will speak to/interact with what.

Reviewer 2: ```suggestion
Give a high-level overview of the tutorial. This helps the reader understand the scope of what they'll be doing end-to-end and potentially the basic architecture of what will speak to/interact with what.
```

> If relevant, include what they'll have achieved or built "by the end of this tutorial." 

Reviewer: I think we can remove the quotation marks.
```suggestion
If relevant, include what they'll have achieved or built by the end of this tutorial.
```

> ## Pre-requisites

Reviewer: ```suggestion
## Prerequisites
```

> This can incluce:

Reviewer: ```suggestion
This can include:
```

- [ ] Recommended hardware/software/local setup
- [ ] Familiarity of experience with Rust, JavaScript, at a certain level
- [ ] Completion of a prior tutorial or process
- [ ] Accounts or logins

> - [ ] Anything else that if relevant

Reviewer: ```suggestion
- [ ] Anything else that is relevant
```

## Steps

### Step 1

Info here.

### Step 2

Info here.

> ### Troubleshooting

Reviewer: It might be good to have this a level higher, as it's not part of the "steps".
```suggestion
## Troubleshooting
```

> Advice on common gotchas or specific error messages that might pop up, or the support route if the user needs help.

Reviewer: I suggest removing slang words
```suggestion
Advice on common questions or specific error messages that might pop up or the support route if the user needs help.
```

> ### Post-tutorial

Reviewer: Maybe we can call this Next Steps or something different, and move it a level higher.
```suggestion
##  Next Steps
```

> Further tutorials or docs to check out, once someone has successfully completed your tutorial.

Reviewer: Use "read" or "look at" instead of "check out". It's a little more formal.
```suggestion
Further tutorials or docs to read once someone has successfully completed your tutorial.
```

> ## Things to remember

Reviewer: ```suggestion
## Things to Remember
```

### Tag your code snippets

> Remember to add the language when including any code blocks. While not supported natively by markdown, many markdown engines (including the one used by GitHub) will support syntax highlighting which is helpful for reviewers and future maintainers of your tutorial:

Reviewer: ```suggestion
Remember to add the language when including any code blocks. While not supported natively by markdown, many markdown engines (including the one used by GitHub) will support syntax highlighting, which is helpful for reviewers and future maintainers of your tutorial:
```

```
    ```rust
        let (contract_hash, contract_version) = 
        storage::add_contract_version(contract_package_hash, 
                                    entry_points, 
                                    named_keys);
    ```
```

> ### Single or multi-page?

Reviewer 2: It might just be a quirk of the Github review window, but for some reason the code snippet above is including the "Single or multi-page" subsection.

I checked in VS Code and the 'Edit File' window, and I don't seem to be getting the same issue.

Author: hmmm ok.. it #looksFineOnMyMachine in VSCode and local docs but let me know if it renders badly on the actual docs when in prod 
<img width="757" alt="image" src="https://user-images.githubusercontent.com/38460768/216675792-8eead288-dd85-414a-ac9b-34578c4474ec.png">

> ### Single or multi-page?

Reviewer: ```suggestion
### Single or multi-page tutorial?
```
